static/app.js

```diff
@@ -6,32 +6,31 @@ $(function () {
 	$('[data-toggle="tooltip"]').tooltip();
 });
 
-let input = '';
 
-function balancedText() {
-	/* Example to count the occurrences of a string in a string.
-		var theString = "This is a string.";
-		console.log(theString.split("is").length - 1);
-		*/
-	var balanced = '✓ Balanced';
-	var unbalanced = '❌ Unbalanced';
-	var opening = input.split('{').length - 1;
-	var closing = input.split('}').length - 1;
-	// Returns balanced if opening is equal to closing, otherwise unbalanced
-	return opening === closing ? balanced : unbalanced;
-}
+function balancedText(input) {
+	const stack = [];
+	
+	for (let i = 0; i < input.length; i++) {
+		const char = input[i];
 
-function charCount() {
-	return input.length;
-}
+		if (char === '{' && input[i - 1] !== '\\') {
+			stack.push('{');
+			continue;
+		}
+		if (char === '}' && input[i - 1] !== '\\' && stack.pop() === undefined) {
+			return false;
+		}
+	}
 
-function lineCount() {
-	return input.split('\n').length;
+	return stack.length === 0;
 }
 
-function copy() {
-	this.input.select();
-	document.execCommand('copy');
+function lineCount(input) {
+	let count = 0;
+	for (const char of input) {
+		count += char === '\n';
+	}
+	return count;

```

Comment:
Should include first line
```suggestion
	return count + 1;
```